core/schema/container.go

```diff
@@ -821,14 +869,39 @@ func (s *containerSchema) build(ctx context.Context, parent *core.Container, arg
 		return nil, err
 	}
 
+	vals := make([]core.SecretArgInternal, len(args.SecretArgs))
+	for i, arg := range args.SecretArgs {
+		secret, ok := secretStore.GetSecret(arg.Value.Value.ID().Digest())
+		if !ok {
+			return nil, fmt.Errorf("secret %q not found", arg.Value.Value.ID().Digest())
+		}
+
+		// if secret name is not explicitly provided, fallback to fetching
+		// from the secret store
+		secretName := arg.Value.Name
+		if arg.Value.Name == "" {
```

Comment:
:eyes: this should only be for the fallback case right? It shouldn't be possible to trigger this by passing an empty string to the new API.

Reply:
The comment here https://github.com/dagger/dagger/issues/7358#issuecomment-2321481240 talked about possibility of having a fallback. 

happy to go either route depending on how folks recommend.

Reply:
Yes there should be a fallback, but we shouldn't allow setting a secret with an empty string, where this would fallback to reading from the secret store.